core/src/main/java/org/testcontainers/containers/wait/HostPortWaitStrategy.java

```diff
@@ -33,7 +33,7 @@ protected void waitUntilReady() {
         Callable<Boolean> checkStrategy;
 
         // Special case for Docker for Mac, see #160
-        if (DockerClientFactory.instance().isUsing(ProxiedUnixSocketClientProviderStrategy.class)) {
+        if (System.getProperty("os.name").toLowerCase().contains("mac")) {
```

Comment:
I don't think that this will break mac compatibility (the usage of `ProxiedUnixSocketClientProviderStrategy` is determined by the same check), but since there are no automated mac tests, playing it safe might be more sensible here?